### PR TITLE
UPSTREAM:<carry> Ignore the topology based labels while comparing nodegroups

### PR DESCRIPTION
**Pull request by hardikdr**
**What this PR does / why we need it**: With k8s 1.17, nodes are labeled with topology-based keywords. With this PR we ignore these labels while comparing the node groups.

**Which issue(s) this PR fixes**:
Partially fixes https://github.com/gardener/autoscaler/issues/31

**Special notes for your reviewer**:

**Release note**:

```improvement operator
Add topology.kubernetes.io labels to be ignored when comparing similar node groups.
```

### CODE REVIEW

File: cluster-autoscaler/utils/nodegroupset/compare_nodegroups.go [hunk 1]
```diff
@@ -124,6 +124,12 @@ func IsNodeInfoSimilar(n1, n2 *schedulercache.NodeInfo) bool {
 			if label == kubeletapis.LabelGardenerWorkerPool {
 				continue
 			}
+			if label == "topology.kubernetes.io/zone" {
+				continue
+			}
+			if label == "topology.kubernetes.io/region" { // TODO: Rebase to latest CA
```
Comment (prashanth26) [inline cluster-autoscaler/utils/nodegroupset/compare_nodegroups.go:130]: Please change these into `StringConstants`. Ideally would be good to have something like this - https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/processors/nodegroupset/compare_nodegroups.go#L42-L50. However, `StringConstants` should be okay.

File: cluster-autoscaler/utils/nodegroupset/compare_nodegroups.go [hunk 2]
```diff
@@ -118,10 +127,19 @@ func IsNodeInfoSimilar(n1, n2 *schedulercache.NodeInfo) bool {
 			if label == kubeletapis.LabelZoneRegion {
 				continue
 			}
-			if label == kubeletapis.LabelGardenerWorkerGroup {
+			if label == LabelGardenerWorkerGroup {
```
Comment (prashanth26) [inline cluster-autoscaler/utils/nodegroupset/compare_nodegroups.go:130]: Why this change? If we change this, we could change `kubeletapis.LabelZoneRegion` also or keep both?

File: cluster-autoscaler/utils/nodegroupset/compare_nodegroups.go [hunk 3]
```diff
@@ -34,7 +34,7 @@ const (
 	MaxFreeDifferenceRatio = 0.05
 	// MaxMemoryDifferenceInKiloBytes describes how much memory
 	// capacity can differ but still be considered equal.
-	MaxMemoryDifferenceInKiloBytes = 128000
+	MaxMemoryDifferenceInKiloBytes = 256000000
```
Comment (hardikdr) [inline cluster-autoscaler/utils/nodegroupset/compare_nodegroups.go:37]: With the experiments with m5.large machines, I saw the following allocatable for 3 nodes with the same machine-type. 
```
{
  "cpu": "2",
  "ephemeral-storage": "50553132Ki",
  "hugepages-1Gi": "0",
  "hugepages-2Mi": "0",
  "memory": "7855172Ki",
  "pods": "110"
}
{
  "cpu": "2",
  "ephemeral-storage": "50553132Ki",
  "hugepages-1Gi": "0",
  "hugepages-2Mi": "0",
  "memory": "7941188Ki",
  "pods": "110"
}
{
  "cpu": "2",
  "ephemeral-storage": "50553132Ki",
  "hugepages-1Gi": "0",
  "hugepages-2Mi": "0",
  "memory": "7855172Ki",
  "pods": "110"
}
```

And code-wise the difference between allocatable.memory for second and third node came out to be: `8.8072192e+07` . This confuses a bit, but autoscaler couldn't consider the 2nd and 3nd node-groups similar, and couldn't balance. I'll take a defensive approach and have a large value to balance the machines.

File: cluster-autoscaler/utils/nodegroupset/compare_nodegroups.go (hunk 1 above)
Comment (hardikdr) [reply cluster-autoscaler/utils/nodegroupset/compare_nodegroups.go]: done. 
Unfortunately, the constants in mcm-branch are coming from the vendored kubelet, and I dint want to update the vendor only to add the topology constants there. The right approach is to re-vendor the k8s version. The link you posted is from the master branch, code-path is different than our fork. 

As I anyways added constants in the current file, also migrating the gardener-specific constants here to nullify the previous change in the vendor.

File: cluster-autoscaler/utils/nodegroupset/compare_nodegroups.go (hunk 2 above)
Comment (hardikdr) [reply cluster-autoscaler/utils/nodegroupset/compare_nodegroups.go:130]: Please see the comment above.

File: cluster-autoscaler/utils/nodegroupset/compare_nodegroups.go (hunk 3 above)
Comment (hardikdr) [reply cluster-autoscaler/utils/nodegroupset/compare_nodegroups.go:37]: https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/processors/nodegroupset/compare_nodegroups.go#L66:1 upstream seems to have fixed this though.

Comment (hardikdr) [reply cluster-autoscaler/utils/nodegroupset/compare_nodegroups.go:37]: I think there are good changes in upstream in terms of comparing labels, better if we fix it while rebasing. Current changes seem to be fine for now.

Comment (prashanth26) [reply cluster-autoscaler/utils/nodegroupset/compare_nodegroups.go:37]: Okay.
